lib/rules/no-broken-super-chain.js

```diff
@@ -3,93 +3,107 @@
  */
 
 const MESSAGES = {
-  noSuper: '\'this._super(...arguments)\' must be called in init()',
-  noThisBeforeSuper: 'Must call \'this._super(...arguments)\' before accessing `this`',
-  tooManySupers: 'Only call `this._super(...arguments)` once per init()'
+  noSuper: '`this._super(...arguments)` must be called in',
+  tooManySupers: 'Only call `this._super(...arguments)` once per lifecycle hook',
+  argsNotPassedToSuper: '...arguments need to be passed to this._super() call'
 };
 
-// TODO: Make this configurable
-const EMBER_MODULES_WITH_SUPER_CHAIN = {
-  Component: true,
-  Mixin: true,
-  Route: true,
-  Controller: true,
-  View: true
-};
+const LIFECYCLE_HOOKS = ['init', 'willDestroy', 'willDestroyElement', 'destroy'];
 
-/**
- * Determines if this is an init method in an extension of Ember[EMBER_MODULES_WITH_SUPER_CHAIN.*]
- * @param {Node} node
- */
-function isInit(node) {
-  if (node.type === 'FunctionExpression' && node.parent && node.parent.key && node.parent.key.name === 'init') {
+function isExtend(node) {
+  return node && node.callee && node.callee.property && node.callee.property.name === 'extend';
+}
+
+function isSuperCall(lineWithinFn) {
```

Comment:
One more case we need to handle is:
```js
Foo.extend({
  init() {
    return this._super(...argument);
  }
});
```
This ^ is valid, the linting rule should not complain about it.

Reply:
Great catch. Updated to: `expression = expressionStatement.type === 'ReturnStatement' ? expressionStatement.argument : expressionStatement.expression;`